src/schneider_lab_to_nwb/schneider_2024/schneider_2024_optogeneticinterface.py

```diff
@@ -0,0 +1,78 @@
+"""Primary class for converting optogenetic stimulation."""
+from pynwb.file import NWBFile
+from pydantic import FilePath
+import numpy as np
+from pymatreader import read_mat
+from pynwb.device import Device
+from pynwb.ogen import OptogeneticSeries, OptogeneticStimulusSite
+
+from neuroconv.basedatainterface import BaseDataInterface
+from neuroconv.utils import DeepDict
+
+
+class Schneider2024OptogeneticInterface(BaseDataInterface):
+    """Optogenetic interface for schneider_2024 conversion"""
+
+    keywords = ["optogenetics"]
+
+    def __init__(self, file_path: FilePath):
+        super().__init__(file_path=file_path)
+
+    def get_metadata(self) -> DeepDict:
+        metadata = super().get_metadata()
+
+        return metadata
+
+    def get_metadata_schema(self) -> dict:
+        metadata_schema = super().get_metadata_schema()
+        return metadata_schema
+
+    def add_to_nwbfile(self, nwbfile: NWBFile, metadata: dict):
+        # Read Data
+        file_path = self.source_data["file_path"]
+        file = read_mat(file_path)
+        onset_times = file["events"]["push"]["opto_time"]
+        is_opto_trial = np.logical_not(np.isnan(onset_times))
+        onset_times = onset_times[is_opto_trial]
+        offset_times = file["events"]["push"]["opto_time_end"]
+        offset_times = offset_times[is_opto_trial]
+        assert np.all(
+            np.logical_not(np.isnan(offset_times))
+        ), "Some of the offset times are nan when onset times are not nan."
+        power = metadata["Optogenetics"]["OptogeneticSeries"]["power"]
+
+        timestamps, data = [0], [0]
+        for onset_time, offset_time in zip(onset_times, offset_times):
+            timestamps.append(onset_time)
+            data.append(power)
+            timestamps.append(offset_time)
+            data.append(0)
```

Comment:
I guess this is an edge case of a TimeSeries that is almost an events object but this way you can also associate it with an ogen site, right?

It makes sense to me. Do you really need the first data point (0 power at time 0)?

Reply:
> I guess this is an edge case of a TimeSeries that is almost an events object but this way you can also associate it with an ogen site, right?

Yeah, not really an edge case tho -- all of the conversions I've worked on with Ogen have stim times and parameters rather than a true time series. It definitely should be updated in NWB, but I'm not sure which extension should include this change.

Reply:
> It makes sense to me. Do you really need the first data point (0 power at time 0)?

Hmmm, I guess not...

Reply:
Edge case in terms of the typing system of the nwb not of the frequency of your use.